lib/IRGen/GenMeta.cpp

```diff
@@ -2149,11 +2194,13 @@ namespace {
     }
 
     void addInstanceSize() {
-      if (llvm::Constant *size
-            = tryEmitClassConstantFragileInstanceSize(IGM, Target)) {
+      if (FieldLayout.isFixedLayout()) {
+        auto *size = IGM.getSize(FieldLayout.getSize());
```

Comment:
The `getTrunc` dance here was only necessary because we were starting with a `Constant*`; if we're starting with a `Size` we might as well just build it in the right type in the first place.